src/shell/commands/executable.rs

```diff
@@ -55,24 +55,44 @@ impl ShellCommand for ExecutableCommand {
       // avoid deadlock since this is holding onto the pipes
       drop(sub_command);
 
-      tokio::select! {
-        result = child.wait() => match result {
-          Ok(status) => ExecuteResult::Continue(
-            status.code().unwrap_or(1),
-            Vec::new(),
-            Vec::new(),
-          ),
-          Err(err) => {
-            let _ = stderr.write_line(&format!("{}", err));
-            ExecuteResult::Continue(1, Vec::new(), Vec::new())
+      loop {
+        tokio::select! {
+          result = child.wait() => match result {
+            Ok(status) => return ExecuteResult::Continue(
+              status.code().unwrap_or(1),
+              Vec::new(),
+              Vec::new(),
+            ),
+            Err(err) => {
+              let _ = stderr.write_line(&format!("{}", err));
+              return ExecuteResult::Continue(1, Vec::new(), Vec::new())
+            }
+          },
+          signal = context.state.kill_signal().wait_any() => {
+            if let Some(_id) = child.id() {
+              #[cfg(unix)]
+              kill(_id as i32, signal);
```

Comment:
Is this on purpose that only unix is supported? Does it work alright on Windows already?

Reply:
I think it tries to kill the child below on Windows (without the correct signal)

Reply:
Basically if the signal is considered as "this should abort stuff" then it will just kill on windows.